File: app_codebase/dart_bkend_base_02/bin/modules/ws/demo_ws_service.dart
/// Demo WebSocket channel handlers — ping/pong and echo.
library;

import '../../core/state/room/room_demo_handler.dart';
import '../../core/ws/contracts/ws_message_contract.dart';

Map<String, dynamic>? handleSystem(
  WsConnectionContext ctx,
  WsClientMessage msg,
) {
  if (msg.msgType == 'ping') {
    return {
      'type': 'pong',
      'channel': 'system',
      'ts': DateTime.now().toUtc().toIso8601String(),
    };
  }
  return null;
}

Map<String, dynamic>? handleDemoEcho(
  WsConnectionContext ctx,
  WsClientMessage msg,
) {
  if (msg.msgType == 'event') {
    final text = msg.payload['text']?.toString() ?? '';
    return {
      'type': 'event',
      'channel': 'demo/echo',
      'payload': {'echo': text},
    };
  }
  return null;
}

Map<String, dynamic>? handleDemoRoom(
  WsConnectionContext ctx,
  WsClientMessage msg,
) =>
    handleDemoRoomMessage(ctx, msg);

File: app_codebase/python_base_05/bin/modules/ws/demo_ws_service.py
"""Demo WebSocket channel handlers — ping/pong and echo."""

from __future__ import annotations

from datetime import datetime, timezone
from typing import Any

from core.ws.contracts.ws_message_contract import WsClientMessage, WsConnectionContext


def handle_system(ctx: WsConnectionContext, msg: WsClientMessage) -> dict[str, Any] | None:
    if msg.msg_type == "ping":
        return {
            "type": "pong",
            "channel": "system",
            "ts": datetime.now(timezone.utc).isoformat(),
        }
    return None


def handle_demo_echo(ctx: WsConnectionContext, msg: WsClientMessage) -> dict[str, Any] | None:
    if msg.msg_type == "event":
        text = str(msg.payload.get("text", ""))
        return {
            "type": "event",
            "channel": "demo/echo",
            "payload": {"echo": text},
        }
    return None
